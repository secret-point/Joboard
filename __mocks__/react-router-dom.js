// Mock navigate and useParams when needed
// const navigate = jest.fn();
// const useParams = jest.fn();
const useLocation = jest.fn();

module.exports = {
  ...jest.requireActual("react-router-dom"),
  // useNavigate: () => navigate,
  // useParams,
  useLocation,
};
