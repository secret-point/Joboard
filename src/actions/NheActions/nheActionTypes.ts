import { Action } from "redux";
import {
  GetNheTimeSlotRequestDs,
  GetNheTimeSlotRequestThroughNheDS,
  GetPossibleNhePreferenceRequest,
  NHETimeSlot,
  PossibleNhePreferenceConfig, SavePossibleNhePreferenceRequest
} from "../../utils/types/common";

export enum NHE_ACTION_TYPES {
  GET_SLOTS_DS = "GET_AVAILABLE_TIME_SLOTS_DS",
  GET_SLOTS_THROUGH_NHE_DS = "GET_AVAILABLE_TIME_SLOTS_THROUGH_NHE_DS",
  GET_SLOTS_DS_SUCCESS = "GET_AVAILABLE_TIME_SLOTS_DS_SUCCESS",
  GET_SLOTS_DS_FAILED = "GET_AVAILABLE_TIME_SLOTS_DS_FAILED",
  GET_POSSIBLE_NHE_PREFERENCES = "GET_POSSIBLE_NHE_PREFERENCES",
  GET_POSSIBLE_NHE_PREFERENCES_SUCCESS = "GET_POSSIBLE_NHE_PREFERENCES_SUCCESS",
  GET_POSSIBLE_NHE_PREFERENCES_FAILED = "GET_POSSIBLE_NHE_PREFERENCES_FAILED",
  SET_POSSIBLE_NHE_PREFERENCE_REQUEST = "SET_POSSIBLE_NHE_PREFERENCE_REQUEST"
}

export interface GetNheTimeSlotsDsAction extends Action {
  type: NHE_ACTION_TYPES.GET_SLOTS_DS;
  payload: GetNheTimeSlotRequestDs;
  onSuccess?: Function;
  onError?: Function;
}

export interface GetNheTimeSlotsThroughNheDsAction extends Action {
  type: NHE_ACTION_TYPES.GET_SLOTS_THROUGH_NHE_DS;
  redirectWhenNoData: boolean;
  payload: GetNheTimeSlotRequestThroughNheDS;
  onSuccess?: Function;
  onError?: Function;
}

export interface GetNheTimeSlotsDsSuccessAction extends Action {
  type: NHE_ACTION_TYPES.GET_SLOTS_DS_SUCCESS;
  payload: NHETimeSlot[];
}

export interface GetNheTimeSlotsDsFailedAction extends Action {
  type: NHE_ACTION_TYPES.GET_SLOTS_DS_FAILED;
  payload: any;
}

export interface GetPossibleNhePreferencesAction extends Action {
  type: NHE_ACTION_TYPES.GET_POSSIBLE_NHE_PREFERENCES;
  payload: GetPossibleNhePreferenceRequest;
}

export interface GetPossibleNhePreferencesSuccessAction extends Action {
  type: NHE_ACTION_TYPES.GET_POSSIBLE_NHE_PREFERENCES_SUCCESS;
  payload: PossibleNhePreferenceConfig;
}

export interface GetPossibleNhePreferencesFailedAction extends Action {
  type: NHE_ACTION_TYPES.GET_POSSIBLE_NHE_PREFERENCES_FAILED;
  payload: any;
}

export interface SetPossibleNhePreferenceRequestAction extends Action {
  type: NHE_ACTION_TYPES.SET_POSSIBLE_NHE_PREFERENCE_REQUEST;
  payload: SavePossibleNhePreferenceRequest;
}

export type NheTimeSlotsActions =
    GetNheTimeSlotsDsAction |
    GetNheTimeSlotsDsSuccessAction |
    GetNheTimeSlotsDsFailedAction |
    GetNheTimeSlotsThroughNheDsAction |
    GetPossibleNhePreferencesAction |
    GetPossibleNhePreferencesSuccessAction |
    GetPossibleNhePreferencesFailedAction |
    SetPossibleNhePreferenceRequestAction;
