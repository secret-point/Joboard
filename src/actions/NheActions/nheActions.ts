import {
  GetNheTimeSlotRequestDs,
  GetNheTimeSlotRequestThroughNheDS,
  GetPossibleNhePreferenceRequest,
  NHETimeSlot,
  PossibleNhePreferenceConfig,
  SavePossibleNhePreferenceRequest
} from "../../utils/types/common";
import {
  GetNheTimeSlotsDsAction,
  GetNheTimeSlotsDsFailedAction,
  GetNheTimeSlotsDsSuccessAction,
  GetNheTimeSlotsThroughNheDsAction,
  GetPossibleNhePreferencesAction,
  GetPossibleNhePreferencesFailedAction,
  GetPossibleNhePreferencesSuccessAction,
  NHE_ACTION_TYPES,
  SetPossibleNhePreferenceRequestAction
} from "./nheActionTypes";

export const actionGetNheTimeSlotsThroughNheDs = (payload: GetNheTimeSlotRequestThroughNheDS, redirectWhenNoData: boolean): GetNheTimeSlotsThroughNheDsAction => {
  return {
    type: NHE_ACTION_TYPES.GET_SLOTS_THROUGH_NHE_DS,
    redirectWhenNoData,
    payload
  };
};

export const actionGetNheTimeSlotsDs = (payload: GetNheTimeSlotRequestDs): GetNheTimeSlotsDsAction => {
  return {
    type: NHE_ACTION_TYPES.GET_SLOTS_DS,
    payload
  };
};

export const actionGetNheTimeSlotsDsSuccess = (payload: NHETimeSlot[]): GetNheTimeSlotsDsSuccessAction => {
  return {
    type: NHE_ACTION_TYPES.GET_SLOTS_DS_SUCCESS,
    payload
  };
};

export const actionGetNheTimeSlotsDsFailed = (payload: any): GetNheTimeSlotsDsFailedAction => {
  return {
    type: NHE_ACTION_TYPES.GET_SLOTS_DS_FAILED,
    payload
  };
};

export const actionGetPossibleNhePreference = (payload: GetPossibleNhePreferenceRequest): GetPossibleNhePreferencesAction => {
  return {
    type: NHE_ACTION_TYPES.GET_POSSIBLE_NHE_PREFERENCES,
    payload
  };
};

export const actionGetPossibleNhePreferenceSuccess = (payload: PossibleNhePreferenceConfig): GetPossibleNhePreferencesSuccessAction => {
  return {
    type: NHE_ACTION_TYPES.GET_POSSIBLE_NHE_PREFERENCES_SUCCESS,
    payload
  };
};

export const actionGetPossibleNhePreferenceFailed = (payload: any): GetPossibleNhePreferencesFailedAction => {
  return {
    type: NHE_ACTION_TYPES.GET_POSSIBLE_NHE_PREFERENCES_FAILED,
    payload
  };
};
export const actionSetPossibleNhePreferenceRequest = (payload: SavePossibleNhePreferenceRequest): SetPossibleNhePreferenceRequestAction => {
  return {
    type: NHE_ACTION_TYPES.SET_POSSIBLE_NHE_PREFERENCE_REQUEST,
    payload
  };
};

