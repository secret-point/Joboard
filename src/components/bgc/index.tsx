/* eslint-disable react/jsx-wrap-multilines */
import React from "react";
import ReactDOM from "react-dom";

const Index = () => {
  return <div>Index</div>;
};

// export default index;

ReactDOM.render(<Index />, document.getElementById("root"));
