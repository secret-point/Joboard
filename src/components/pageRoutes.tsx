export enum PAGE_ROUTES {
  DEFAULT = "/",
  CONSENT = "consent",
  PRE_CONSENT = "pre-consent",
  JOB_OPPORTUNITIES = "job-opportunities",
  JOB_CONFIRMATION = "job-opportunities/job-confirmation",
  JOB_DESCRIPTION = "job-opportunities/job-description",
  CONTINGENT_OFFER = "contingent-offer",
  BACKGROUND_CHECK = "bgc",
  BACKGROUND_CHECK_FCRA = "bgc/fcra",
  RESUME_APPLICATION = "resume-application",
  NHE = "nhe",
  ALREADY_APPLIED = "already-applied",
  SELF_IDENTIFICATION = "self-identification",
  REVIEW_SUBMIT = "review-submit",
  THANK_YOU = "thank-you",
  SESSION_TIMEOUT = "session-timeout",
  WOTC = "wotc",
  WOTC_COMPLETE = "wotc-complete",
  ASSESSMENT = "assessment",
  ASSESSMENT_CONSENT = "assessment-consent",
  ASSESSMENT_NOT_ELIGIBLE = "assessment-not-eligible",
  ASSESSMENT_FINISHED = "assessment-finished",
  CANDIDATE_WITHDRAWS = "candidate-withdraws",
  CALI_DISCLOSURE = "cali-disclosure",
  AMAZON_REJECTS = "amazon-rejects",
  WORKFLOW_FAILED = "workflow-failed",
  AMAZON_WITHDRAWS = "amazon-withdraws",
  NO_AVAILABLE_TIME_SLOTS = "no-available-time-slots",
  CAN_NOT_OFFER_JOB = "can-not-offer-job",
  SUPPLEMENTARY_SUCCESS = "supplementary-success",
  REHIRE_ELIGIBILITY_STATUS = "rehire-eligibility-status",
  NO_AVAILABLE_SHIFT = "no-available-shift",
  APPLICATIONID_NULL = "applicationId-null",
  SHIFT_PREFERENCE = "job-opportunities/shift-preferences",
  ADDITIONAL_INFORMATION = "additional-information"
}

// Reference: https://code.amazon.com/packages/HVHStepFunctionDefinitions/blobs/ad75b9869a017bc84a9e03920e835a6b2cec7063/--/configuration/nacf_tier_1_node_workflow/defintion.json#L186
export const PagesControlledByWorkFlowService = [
  PAGE_ROUTES.JOB_OPPORTUNITIES,
  PAGE_ROUTES.ASSESSMENT_CONSENT,
  PAGE_ROUTES.ASSESSMENT,
  PAGE_ROUTES.CONTINGENT_OFFER,
  PAGE_ROUTES.BACKGROUND_CHECK,
  PAGE_ROUTES.CAN_NOT_OFFER_JOB,
  PAGE_ROUTES.CALI_DISCLOSURE,
  PAGE_ROUTES.NHE,
  PAGE_ROUTES.SELF_IDENTIFICATION,
  PAGE_ROUTES.REVIEW_SUBMIT,
  PAGE_ROUTES.THANK_YOU,
  PAGE_ROUTES.WOTC,
  PAGE_ROUTES.SUPPLEMENTARY_SUCCESS,
  PAGE_ROUTES.ASSESSMENT_NOT_ELIGIBLE,
  PAGE_ROUTES.REHIRE_ELIGIBILITY_STATUS,
  PAGE_ROUTES.AMAZON_WITHDRAWS,
  PAGE_ROUTES.CANDIDATE_WITHDRAWS,
  PAGE_ROUTES.AMAZON_REJECTS,
  PAGE_ROUTES.WORKFLOW_FAILED,
  PAGE_ROUTES.RESUME_APPLICATION,
  PAGE_ROUTES.SHIFT_PREFERENCE,
  PAGE_ROUTES.ADDITIONAL_INFORMATION
];
