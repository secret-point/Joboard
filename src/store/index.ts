import { createStore, applyMiddleware, compose  } from "redux";
import thunk from "redux-thunk";
import rootReducer from "../reducers";
import { createHashHistory } from "history";
import { routerMiddleware  } from "react-router-redux";
import { createLogger } from "redux-logger";

declare global {
  interface Window {
    __REDUX_DEVTOOLS_EXTENSION_COMPOSE__?: typeof compose;
  }
}

const composeEnhancers = (typeof window !== 'undefined' && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) || compose;

const middleware = [thunk, routerMiddleware(createHashHistory()), createLogger()];
const store = createStore(rootReducer, {} , composeEnhancers(applyMiddleware(...middleware)));
export default store;