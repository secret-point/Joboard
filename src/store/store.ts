import { applyMiddleware, createStore } from "redux";
import { routerMiddleware } from "connected-react-router";
import { createHashHistory } from "history";
import { createEpicMiddleware } from "redux-observable";
import createRootReducer from "../reducers/root.reducer";
import rootEpic from "../epics/root.epic";
import { composeWithDevTools } from "redux-devtools-extension";
import { persistReducer, persistStore } from 'redux-persist';
import storage from 'redux-persist/lib/storage'

/* TODO remove any from functions */
const logger = () => (next: any) => (action: any) => {
  // eslint-disable-next-line no-console
  // console.log("dispatching", action);
  const result = next(action);
  // console.log("next state", store.getState());
  // eslint-disable-next-line no-console
  return result;
};

export const history = createHashHistory();

const persistConfig = {
  key: 'root',
  storage, // defaults to localStorage for web
  whitelist: ['request']
};

export const configureStore = (preloadedState: any) => {
  const epicMiddleware = createEpicMiddleware();
  const middlewares: any = [epicMiddleware, routerMiddleware(history), logger];
  const middlewareEnhancer = applyMiddleware(...middlewares);

  const enhancers = [middlewareEnhancer];
  const composedEnhancers: any = composeWithDevTools(...enhancers);
  const persistedReducer = persistReducer(persistConfig, createRootReducer(history));

  const store = createStore(
    persistedReducer,
    preloadedState,
    composedEnhancers
  );

  epicMiddleware.run(rootEpic);

  return store;
};

const store = configureStore({});

export const persistor = persistStore(store);

export default store;
