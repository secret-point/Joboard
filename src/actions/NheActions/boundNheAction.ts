import store from "../../store/store";
import { GetNheTimeSlotRequestDs, GetNheTimeSlotRequestThroughNheDS } from "../../utils/types/common";
import { actionGetNheTimeSlotsDs, actionGetNheTimeSlotsThroughNheDs } from "./nheActions";

export const boundGetNheTimeSlotsDs = (payload: GetNheTimeSlotRequestDs) =>
  store.dispatch(actionGetNheTimeSlotsDs(payload));

export const boundGetNheTimeSlotsThroughNheDs = (payload: GetNheTimeSlotRequestThroughNheDS) =>
  store.dispatch(actionGetNheTimeSlotsThroughNheDs(payload));
