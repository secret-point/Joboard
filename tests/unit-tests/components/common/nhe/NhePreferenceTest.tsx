import { shallow } from "enzyme";
import React from "react";
import NhePreferenceWithStore, { NhePreferences } from "../../../../../src/components/common/nhe/NhePreferences";
import { mountWithStencil } from "@amzn/stencil-react-components/tests";
import { PAGE_ROUTES } from "../../../../../src/components/pageRoutes";
import {
  TEST_APPLICATION_DATA,
  TEST_APPLICATION_ID,
  TEST_APPLICATION_STATE,
  TEST_CANDIDATE_STATE,
  TEST_JOB,
  TEST_JOB_ID,
  TEST_JOB_STATE,
  TEST_NHE_STATE,
  TEST_SCHEDULE_ID,
  TestSavePossibleNheDateRequest
} from "../../../../test-utils/test-data";
import store from "../../../../../src/store/store";
import { Provider } from "react-redux";
import { useLocation } from "react-router-dom";
import * as nheBoundAction from "../../../../../src/actions/NheActions/boundNheAction";

const boundSetPossibleNhePreferenceRequestSpy = jest.spyOn(nheBoundAction, "boundSetPossibleNhePreferenceRequest");
describe("NhePreference Component", () => {
  const initStoreState = store.getState();
  let state: any;
  store.getState = () => state;

  const mockLocation = {
    pathname: "/nhe/nhe-preference",
    search: `?jobId=${TEST_JOB_ID}&scheduleId=${TEST_SCHEDULE_ID}&applicationId=${TEST_APPLICATION_ID}`,
    hash: "",
    state: null
  };
  const mockUseLocation = useLocation as jest.Mock;

  beforeEach(() => {
    state = {
      ...initStoreState,
      application: TEST_APPLICATION_STATE,
      job: TEST_JOB_STATE,
      candidate: TEST_CANDIDATE_STATE,
      nhe: TEST_NHE_STATE
    };
    mockUseLocation.mockReturnValue(mockLocation);
    boundSetPossibleNhePreferenceRequestSpy.mockReset();
  });

  it("should match snapshot", () => {
    const shallowWrapper = shallow(
      <NhePreferences job={TEST_JOB} application={TEST_APPLICATION_DATA} candidate={TEST_CANDIDATE_STATE} nhe={TEST_NHE_STATE} />);

    expect(shallowWrapper).toMatchSnapshot();
  });

  test("it should redirect to Nhe when click to back to shift selection link", () => {
    window.location.hash = `#/${PAGE_ROUTES.NHE_PREFERENCES}`;
    const wrapper = mountWithStencil(
      <Provider store={store}>
        <NhePreferenceWithStore />
      </Provider>);
    const goBackToShiftSelectionButton = wrapper.find("Row.nhePreferenceLink");
    goBackToShiftSelectionButton.simulate("click");

    wrapper.update();
    expect(window.location.hash).toEqual(`#/${PAGE_ROUTES.NHE}`);
  });

  it("should display error if save preference button is clicked with invalid input", () => {
    let wrapper = mountWithStencil(
      <Provider store={store}>
        <NhePreferenceWithStore />
      </Provider>);
    const saveShiftPreference = wrapper.find("Button#saveNhePreference");
    saveShiftPreference.simulate("click");

    wrapper = wrapper.update();
    expect(wrapper.find("InputFooter#nheDateErrorContainer").text()).toContain("Please select at least one date.");
    expect(wrapper.find("InputFooter#nheTimeSlotErrorContainer").text()).toContain("Please select at least one time range.");
  });

  it("should update request if select Location input change ", () => {
    const wrapper = mountWithStencil(
      <Provider store={store}>
        <NhePreferenceWithStore />
      </Provider>);

    const nheSelectLocationInput = wrapper.find("select#nheSelectLocationInput");
    nheSelectLocationInput.simulate("change", { target: {
      label: "Templeton bridge",
      value: "Templeton bridge",
      checked: false
    } });

    expect(boundSetPossibleNhePreferenceRequestSpy).toHaveBeenCalledWith({
      ...TestSavePossibleNheDateRequest,
      possibleCities: [
        {
          ...TestSavePossibleNheDateRequest.possibleCities[0],
          checked: true
        },
        TestSavePossibleNheDateRequest.possibleCities[1]
      ]
    });
  });
});