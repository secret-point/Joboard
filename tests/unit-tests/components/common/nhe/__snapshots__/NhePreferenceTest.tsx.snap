// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NhePreference Component should match snapshot 1`] = `
<Col
  gridGap={10}
>
  <Row
    padding={
      Object {
        "top": "S400",
      }
    }
  >
    <Row
      className="nhePreferenceLink"
      gridGap={8}
      onClick={[Function]}
    >
      <IconArrowLeft
        aria-hidden={true}
        display="block"
        fontSize="T100"
        size={24}
      />
      <Text
        font="primary"
        fontSize="T200"
        fontWeight="medium"
        hyphens="none"
      >
        Back
      </Text>
    </Row>
  </Row>
  <Col
    gridGap={8}
    padding={
      Object {
        "top": "S300",
      }
    }
  >
    <Col
      gridGap={20}
    >
      <H4
        color="neutral90"
        font="primary"
        fontSize="T400"
        fontWeight="regular"
        hyphens="none"
        margin={0}
      >
        Specify your pre-hire appointment availability
      </H4>
      <Text
        font="primary"
        fontSize="T300"
        hyphens="none"
      >
        You can select a location, date and time frame that you are available on. We will reach out to you within 1 business day confirming your appointment. Please arrive at the pre-hire event location on date and time specified in appointment confirmation communication. We look forward to seeing you soon.
      </Text>
    </Col>
    <Col
      gridGap="S300"
      padding={
        Object {
          "top": "S200",
        }
      }
    >
      <Label
        color="inherit"
        font="primary"
        fontSize="T200"
        fontWeight="regular"
        htmlFor="nheLocationSelectInput"
        hyphens="none"
      >
        Location
         *
      </Label>
      <Select
        collapsedOptionsExpanderTitle="Selected items"
        customSelectionOptionsLabel="Group selection options"
        defaultOptionsLabel="Single selection options"
        disabledTagsHiddenText="The options in this list are not dismissible because the input is disabled"
        getTriggerText={[Function]}
        id="nheSelectLocationInput"
        layout="default"
        noOptionsText="No options available"
        onChange={[Function]}
        options={
          Array [
            Object {
              "checked": false,
              "label": "Templeton bridge",
              "value": "Templeton bridge",
            },
            Object {
              "checked": false,
              "label": "Test city",
              "value": "Test city",
            },
          ]
        }
        placeholder="Choose an option"
        renderOption={[Function]}
        searchFieldAriaLabel="Start typing to filter out search results"
        selectedOptionAltText="selected"
        valueAccessor="value"
        width="100%"
      />
    </Col>
  </Col>
  <Col
    gridGap={10}
    padding={
      Object {
        "top": "S300",
      }
    }
  >
    <fieldset
      className="borderlessFieldset"
    >
      <legend>
        <Text
          color="inherit"
          font="primary"
          fontSize="T200"
          hyphens="none"
        >
          Date
           *
        </Text>
      </legend>
      <FormWrapper>
        <InputWrapper
          error={false}
          id="nhePreferenceDate-Wednesday, Jan 25th 2023"
          key="Wednesday, Jan 25th 2023"
          labelPosition="trailing"
          labelText="Wednesday, Jan 25th 2023"
          required={false}
          shouldValidateNonInclusiveTerms={false}
          warning={false}
        >
          <Component />
        </InputWrapper>
      </FormWrapper>
    </fieldset>
  </Col>
  <Col
    gridGap={10}
    padding={
      Object {
        "top": "S300",
      }
    }
  >
    <fieldset
      className="borderlessFieldset"
    >
      <legend>
        <Text
          color="inherit"
          font="primary"
          fontSize="T200"
          hyphens="none"
        >
          Time
           *
        </Text>
      </legend>
      <FormWrapper>
        <InputWrapper
          error={false}
          id="nhePreferenceTimeRange-6:00 AM - 10:00 AM"
          key="6:00 AM - 10:00 AM"
          labelPosition="trailing"
          labelText="6:00 AM - 10:00 AM"
          required={false}
          shouldValidateNonInclusiveTerms={false}
          warning={false}
        >
          <Component />
        </InputWrapper>
        <InputWrapper
          error={false}
          id="nhePreferenceTimeRange-10:00 AM - 2:00 PM"
          key="10:00 AM - 2:00 PM"
          labelPosition="trailing"
          labelText="10:00 AM - 2:00 PM"
          required={false}
          shouldValidateNonInclusiveTerms={false}
          warning={false}
        >
          <Component />
        </InputWrapper>
        <InputWrapper
          error={false}
          id="nhePreferenceTimeRange-2:00 PM - 6:00 PM"
          key="2:00 PM - 6:00 PM"
          labelPosition="trailing"
          labelText="2:00 PM - 6:00 PM"
          required={false}
          shouldValidateNonInclusiveTerms={false}
          warning={false}
        >
          <Component />
        </InputWrapper>
        <InputWrapper
          error={false}
          id="nhePreferenceTimeRange-6:00 PM - 10:00 PM"
          key="6:00 PM - 10:00 PM"
          labelPosition="trailing"
          labelText="6:00 PM - 10:00 PM"
          required={false}
          shouldValidateNonInclusiveTerms={false}
          warning={false}
        >
          <Component />
        </InputWrapper>
      </FormWrapper>
    </fieldset>
  </Col>
  <Col
    gridGap={20}
    padding={
      Object {
        "top": "S400",
      }
    }
  >
    <Button
      colorScheme="default"
      iconPosition="leading"
      id="saveNhePreference"
      onClick={[Function]}
      shouldUseResponsiveContent={true}
      size="default"
      style={
        Object {
          "width": "100%",
        }
      }
      type="button"
      variant="primary"
    >
      Save Preferences
    </Button>
    <Row
      alignItems="center"
      alignSelf="center"
      gridGap={5}
    >
      <Text
        font="primary"
        fontSize="T200"
        hyphens="none"
        textAlign="center"
      >
        If you would like to go back to the previous page and select an available slot
        ,
      </Text>
      <Row
        className="nhePreferenceBackLink"
        onClick={[Function]}
      >
        <Text
          font="primary"
          fontSize="T300"
          hyphens="none"
        >
          click here.
        </Text>
      </Row>
    </Row>
  </Col>
</Col>
`;
